modin/core/storage_formats/base/query_compiler.py

```diff
@@ -137,7 +130,20 @@ def default_to_pandas(self, pandas_op, *args, **kwargs):
     # some of these abstract methods, but for the sake of generality they are
     # treated differently.
 
-    lazy_execution = False
+    @property
+    def lazy_execution(self):
+        """
+        Whether underlying execution engine is designed to be executed in a lazy mode only.
+
+        If True, such QueryCompiler will be handled differently at the front-end in order
+        to reduce execution triggering as much as possible.
```

Comment:
This docstring is no longer true compared to what we're seeing in the implementation; plus I would rather see this change in PandasQueryCompiler, not BaseQueryCompiler (and leave this as a plain attribute in other places).